conformance/08_manifest_delete_test.go

```diff
@@ -11,28 +11,33 @@ import (
 
 var test08ManifestDelete = func() {
 	g.Context("Manifest Delete", func() {
-		g.Specify("DELETE request to manifest URL (tag) MUST fail", func() {
+		g.Specify("DELETE request to manifest should return 202, 400, or 405", func() {
 			req := client.NewRequest(reggie.DELETE, "/v2/<name>/manifests/<reference>",
 				reggie.WithReference(firstTag))
 			resp, err := client.Do(req)
 			Expect(err).To(BeNil())
-			Expect(resp.StatusCode()).To(BeNumerically(">=", 400))
-			Expect(resp.StatusCode()).To(BeNumerically("<", 500))
+			Expect(resp.StatusCode()).To(SatisfyAny(
+				Equal(http.StatusBadRequest),
+				Equal(http.StatusMethodNotAllowed),
+				Equal(http.StatusAccepted)))
+			if resp.StatusCode() == http.StatusBadRequest {
+				errorResponses, err := resp.Errors()
+				Expect(err).To(BeNil())
+				Expect(errorResponses).ToNot(BeEmpty())
+				Expect(errorResponses[0].Code).To(Equal(errorCodes[UNSUPPORTED]))
+			} else if resp.StatusCode() == http.StatusAccepted {
+				deleteByTagAccepted = true
+			}
 		})
 
-		g.Specify("GET request to manifest URL should reveal that delete failed", func() {
-			req := client.NewRequest(reggie.GET, "/v2/<name>/manifests/<digest>", reggie.WithDigest(manifestDigest)).
-				SetHeader("Accept", "application/vnd.oci.image.manifest.v1+json")
-			resp, err := client.Do(req)
-			Expect(err).To(BeNil())
-			Expect(resp.StatusCode()).To(Equal(http.StatusOK))
-		})
-
-		g.Specify("DELETE request to manifest URL (digest) should yield 202 response", func() {
+		g.Specify("DELETE request to manifest URL (digest) should yield 202 response unless already deleted", func() {
 			req := client.NewRequest(reggie.DELETE, "/v2/<name>/manifests/<digest>", reggie.WithDigest(manifestDigest))
 			resp, err := client.Do(req)
 			Expect(err).To(BeNil())
-			Expect(resp.StatusCode()).To(Equal(http.StatusAccepted))
+			// In the case that the previous request was accepted, this may or may not fail (which is ok)
+			if !deleteByTagAccepted {
```

Comment:
I think this is trying to codify too much logic here. Whether the first succeeded shouldn't affect this one. Once we clarify delete as optional, then can just do a similar check as above.

Reply:
@dmcgowan thanks; I just pushed a version that doesn't attempt to codify that logic.  The test in question now allows the same range of response codes, plus 404 to cover the case that delete succeeded.